internal/transactions/services/transaction_service.go

```diff
@@ -134,16 +134,53 @@ func (t *transactionService) BuildAndSignTransactionWithChannelAccount(ctx conte
 		return nil, fmt.Errorf("getting ledger sequence for channel account public key %q: %w", channelAccountPublicKey, err)
 	}
 
+	// Start with wallet backend's default timeout
+	walletBackendTimeBounds := txnbuild.NewTimeout(timeoutInSecs)
+
+	// Smart preconditions merging: use more restrictive TimeBounds if client provides them
+	finalPreconditions := txnbuild.Preconditions{
+		TimeBounds: walletBackendTimeBounds,
+	}
+
+	// If client provided preconditions, merge them intelligently
+	if clientPreconditions.TimeBounds.MaxTime != 0 {
+		// Use the more restrictive time bounds (earlier maxTime wins)
+		if clientPreconditions.TimeBounds.MaxTime < walletBackendTimeBounds.MaxTime {
+			finalPreconditions.TimeBounds.MaxTime = clientPreconditions.TimeBounds.MaxTime
+		}
+	}
+
+	// Use client minTime if provided
+	if clientPreconditions.TimeBounds.MinTime != 0 {
+		finalPreconditions.TimeBounds.MinTime = clientPreconditions.TimeBounds.MinTime
+	}
+
+	// Apply other client preconditions constraints
+	if clientPreconditions.LedgerBounds != nil {
+		finalPreconditions.LedgerBounds = clientPreconditions.LedgerBounds
+	}
+	if clientPreconditions.MinSequenceNumber != nil {
+		finalPreconditions.MinSequenceNumber = clientPreconditions.MinSequenceNumber
+	}
+	if clientPreconditions.MinSequenceNumberAge != 0 {
+		finalPreconditions.MinSequenceNumberAge = clientPreconditions.MinSequenceNumberAge
+	}
+	if clientPreconditions.MinSequenceNumberLedgerGap != 0 {
+		finalPreconditions.MinSequenceNumberLedgerGap = clientPreconditions.MinSequenceNumberLedgerGap
+	}
+	if len(clientPreconditions.ExtraSigners) > 0 {
+		finalPreconditions.ExtraSigners = clientPreconditions.ExtraSigners
+	}
```

Comment:
Do you need to check for empty values before assigning it? I believe the result is the same if you don't check.